### PR TITLE
chore(payment): PAYPAL-2968 e2e tests to cover for moved googlepay-braintree package

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

e2e tests

## Why?

To cover for moved **googlepay-braintree** package

## Testing / Proof

Screenshot 2023-09-22 at 14 18 46

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    braintreePayPal,
+    braintreePayPalCredit,
+    googlePay,
+    internalOrder,
+    order222,
+    orderPayment,
+    payment,
+} from './GooglePayBraintreeMockingResponses';
+
+test.describe('Sample Test Group', () => {
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts:14]: Well, it's not a sample anymore. 😂

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
...W50cmVlZ2F0ZXdheS5jb206NDQzL21lcmNoYW50cy9oZ2RjZG5xNGM2Mnl5YzVyL2NsaWVudF9hcGkiLCJlbnZpcm9ubWVudCI6InNhbmRib3giLCJtZXJjaGFudElkIjoiaGdkY2RucTRjNjJ5eWM1ciIsImFzc2V0c1VybCI6Imh0dHBzOi8vYXNzZXRzLmJyYWludHJlZWdhdGV3YXkuY29tIiwiYXV0aFVybCI6Imh0dHBzOi8vYXV0aC52ZW5tby5zYW5kYm94LmJyYWludHJlZWdhdGV3YXkuY29tIiwidmVubW8iOiJvZmYiLCJjaGFsbGVuZ2VzIjpbImN2diIsInBvc3RhbF9jb2RlIl0sInRocmVlRFNlY3VyZUVuYWJsZWQiOnRydWUsImFuYWx5dGljcyI6eyJ1cmwiOiJodHRwczovL29yaWdpbi1hbmFseXRpY3Mtc2FuZC5zYW5kYm94LmJyYWludHJlZS1hcGkuY29tL2hnZGNkbnE0YzYyeXljNXIifSwicGF5cGFsRW5hYmxlZCI6dHJ1ZSwicGF5cGFsIjp7ImJpbGxpbmdBZ3JlZW1lbnRzRW5hYmxlZCI6dHJ1ZSwiZW52aXJvbm1lbnROb05ldHdvcmsiOmZhbHNlLCJ1bnZldHRlZE1lcmNoYW50IjpmYWxzZSwiYWxsb3dIdHRwIjp0cnVlLCJkaXNwbGF5TmFtZSI6IkJpZ2NvbW1lcmNlIiwiY2xpZW50SWQiOiJBVnZYN0dwTzJELTJocUdaNW9fZWNJVHRxVTd3X1F1QTkxVHQyY29XWTNGdEhIX0JiWXdPZ3lwQmR5LVpFUmhKc1kwUTVMSkZFMm5ma3J4NiIsImJhc2VVcmwiOiJodHRwczovL2Fzc2V0cy5icmFpbnRyZWVnYXRld2F5LmNvbSIsImFzc2V0c1VybCI6Imh0dHBzOi8vY2hlY2tvdXQucGF5cGFsLmNvbSIsImRpcmVjdEJhc2VVcmwiOm51bGwsImVudmlyb25tZW50Ijoib2ZmbGluZSIsImJyYWludHJlZUNsaWVudElkIjoibWFzdGVyY2xpZW50MyIsIm1lcmNoYW50QWNjb3VudElkIjoiYmlnY29tbWVyY2UiLCJjdXJyZW5jeUlzb0NvZGUiOiJVU0QifX0=","returnUrl":null}';
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:17]: One thing we really don't like about mocking is exposing security details like token.
Could you please double check if we can replace tokens in this file with `*` or at least we should use a dummy one?

File: packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts (hunk 1 above)
Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts]: Thanks! Test description has been renamed)

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts (hunk 2 above)
Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts]: @bc-peng Yes we can. Replaced to dummy one. Thanks!

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    '[{"id":"googlepaybraintree","gateway":null,"logoUrl":"","method":"googlepay","supportedCards":["VISA","AMEX","MC"],"providesShippingAddress":true,"config":{"displayName":"Google Pay","cardCode":null,"helpText":"","enablePaypal":null,"merchantId":null,"is3dsEnabled":null,"testMode":true,"isVisaCheckoutEnabled":null,"requireCustomerCode":false,"isVaultingEnabled":false,"isVaultingCvvEnabled":null,"hasDefaultStoredInstrument":false,"isHostedFormEnabled":true,"logo":null},"type":"PAYMENT_TYPE_API","initializationStrategy":{"type":"not_applicable"},"nonce":null,"initializationData":{"card_information":{"type":"Visa","number":"1111","bin":"1234"},"nonce":"7b79202e-c64d-148c-066f-b8b54c8b603a","gateway":"braintree","platformToken":"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJtZXJjaGFudE9yaWdpbiI6Im15LWRldi1zdG9yZS0xOTk5ODYxOTMuc3RvcmUuYmNkZXYiLCJtZXJjaGFudElkIjoiMTU1NDAxNjgwNTE3MzY4MTcyMTAiLCJpYXQiOjE2OTUyOTI4MzF9.qMhongMyAmtokwVfEIC27AUHWORus305phDRjx-ohUEFNEVx0ad7U_zoCmP9WhuT1TvNIKq0VZgEqGcQAF5-Hw","googleMerchantId":"15540168051736817210","googleMerchantName":"My Dev Store 199986193","isThreeDSecureEnabled":false},"clientToken":"clientToken=","returnUrl":null}]';
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:11]: `eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJtZXJjaGFudE9yaWdpbiI6Im15LWRldi1zdG9yZS0xOTk5ODYxOTMuc3RvcmUuYmNkZXYiLCJtZXJjaGFudElkIjoiMTU1NDAxNjgwNTE3MzY4MTcyMTAiLCJpYXQiOjE2OTUyOTI4MzF9.qMhongMyAmtokwVfEIC27AUHWORus305phDRjx-ohUEFNEVx0ad7U_zoCmP9WhuT1TvNIKq0VZgEqGcQAF5-Hw`
Another giant token here.

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const googlePay =
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:13]: Same.

File: packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                headers: { Token: 'Go home, get ahead, light-speed internet' },
+                body: internalOrder,
+            });
+        });
+
+        // Playwright actions
+        await checkout.goto();
+        await checkout.selectPaymentMethod('googlepaybraintree');
+        await checkout.placeOrder();
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts:77]: I believe if a shopper starts at the payment step as a guest, the shopper needs to sign in Google to use the Google Pay.
This test skips that part, which means we don't test this method in full.
The sample Google Pay test for the customer step starts by clicking the Google Pay button. 
Can we follow it to test like a real user as much as we can?

Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts:77]: I see, thanks for the point! Working on it

Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts]: @bc-peng take a look pls. i have updated e2e test

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        directBaseUrl: null,
+        environment: 'offline',
+        braintreeClientId: 'braintreeClientId',
+        merchantAccountId: 'bigcommerce',
+        currencyIsoCode: 'USD',
+    },
+};
+
+const clientToken = btoa(unescape(encodeURIComponent(clientConfig)))
+ */
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:46]: What's this giant comment block for?

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+export const orderPayment =
+    '{"status":"ok","three_ds_result":{"acs_url":null,"payer_auth_request":null,"merchant_data":null,"callback_url":null},"errors":[]}';
+
+export const payment =
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:56]: Is the `clientToken` long string necessary?

File: packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            './packages/test-framework/src/support/orderConfirmation.ejs',
+            { orderId: '222' },
+        );
+
+        await checkout.route('**/checkout.php', `${__dirname}/support/checkout.php.ejs`);
+
+        await checkout.route(
+            'https://pay.google.com/gp/p/js/pay.js',
+            `${__dirname}/support/googlePay.mock.js`,
+        );
```
Comment (bc-peng) [inline packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts:41]: `$__dirname/support/` does not appear to exist here.

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts (hunk 7 above)
Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:56]: This is necessary for correct initialization of the braintree client without errors. If we do not pass the correct `clientToken`, we will encounter the corresponding error and see the error modal window at initialization. It interferes with the success of the test.

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts (hunk 6 above)
Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:46]: To understand what a clientToken is

File: packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts (hunk 8 above)
Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts:41]: Why? It is exist here

Screenshot 2023-10-06 at 10 59 12

Comment (bc-peng) [reply packages/google-pay-integration/e2e/GooglePayBraintreeInPaymentStep.spec.ts:41]: Oh, I see. I didn't notice `$__dirname/support/` was already in the Google Pay package.

File: packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts (hunk 6 above)
Comment (bc-peng) [reply packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts:46]: I am afraid talking about token generation openly is not a good idea.
Could you please move it to the internal confluence and link to it here?
The rest of the PR looks good to me.

Comment (bc-nick) [reply packages/google-pay-integration/e2e/GooglePayBraintreeMockingResponses.ts]: @bc-peng done

**Review by bc-peng — APPROVED** [review]
Nicely done. 👏

**Pull request merged**